app/src/main/java/com/lagradost/cloudstream3/utils/SnackbarHelper.kt

```diff
@@ -0,0 +1,69 @@
+package com.lagradost.cloudstream3.utils
+
+import android.app.Activity
+import android.view.View
+import androidx.annotation.MainThread
+import androidx.annotation.StringRes
+import com.google.android.material.snackbar.Snackbar
+import com.lagradost.api.Log
+import com.lagradost.cloudstream3.R
+import com.lagradost.cloudstream3.mvvm.logError
+import com.lagradost.cloudstream3.utils.UIHelper.colorFromAttribute
+
+object SnackbarHelper {
+
+    private const val TAG = "COMPACT"
+    private var currentSnackbar: Snackbar? = null
+
+    @MainThread
+    fun showSnackbar(
+        act: Activity?,
+        @StringRes message: Int,
+        duration: Int = Snackbar.LENGTH_SHORT,
+        actionText: String? = null,
+        actionCallback: (() -> Unit)? = null
+    ) {
+        if (act == null) return
+        showSnackbar(act, act.getString(message), duration, actionText, actionCallback)
+    }
+
+    @MainThread
+    fun showSnackbar(

```

Comment:
I would rather that this takes a UiText, as then the caller does not need to worry about context

Reply:
> I would rather that this takes a UiText, as then the caller does not need to worry about context

@fire-light42 

I'm confused, I just based this class on the showToast methods in CommonActivity but wanted to avoid adding a lot more to CommonActivity so made its own helper. I just added the methods we actually used and allowed to pass an activity directory (which we could still get from CommonActivity). So maybe I am misunderstanding what you mean here?

Reply:
Instead of passing string as message and actionText you pass UiText like https://github.com/recloudstream/cloudstream/blob/master/app/src/main/java/com/lagradost/cloudstream3/CommonActivity.kt#L118 as that moves the getString or format to the callee from the caller. showToast implements all 3 different methods for convenience.

Reply:
@fire-light42 I *think* I did what you meant? Was still a bit confused, apologies if I did it wrong...